### PR TITLE
Tests for generation with LoRA

### PR DESCRIPTION
**Pull request by Andrei-Aksionov**
Essentially a copycat of tests for base generation and generation with an adapter, but for LoRA.

Includes two specific to LoRA tests:
1. checks that LoRA's global variables are specified.
2. checks that LoRA is applied to at least one layer (otherwise what's the point of running generation with LoRA).

Ideally to check that the instance of Config class inside `generate/lora.py:main` contains lora-specific arguments, but couldn't find how to mock init method of Config class, similar to:
https://github.com/Lightning-AI/lit-gpt/blob/21a6d9ee1884998ef4ed500b89d5bd3a5445983d/tests/test_generate.py#L30-L36
(if it's even possible).

### CODE REVIEW

File: generate/lora.py [hunk 1]
```diff
@@ -85,8 +85,9 @@ def main(
             to_projection=lora_projection,
             to_mlp=lora_mlp,
             to_head=lora_head,
-            **json.load(fp),
```
Comment (Andrei-Aksionov) [inline generate/lora.py:88]: If `lit_config.json` overrides default parameters:
> TypeError: type object got multiple values for keyword argument 'to_query'

File: tests/test_generate_lora.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+from unittest import mock
+from unittest.mock import Mock, call, ANY
+
+import pytest
+import torch
+
+
+@pytest.mark.parametrize("max_seq_length", (10, 5 + 20))
+def test_generate(max_seq_length):
+    import generate.lora as generate
```
Comment (carmocca) [inline tests/test_generate_lora.py:16]: generate.lora is exactly generate.base. Making this test redundant. There's no lora-specific generation code

File: tests/test_generate_lora.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        generate_mock.mock_calls
+        == [call(ANY, ANY, ANY, max_seq_length=101, temperature=2.0, top_k=2, eos_id=ANY)] * num_samples
+    )
+    # only the generated result is printed to stdout
+    assert out.getvalue() == "foo bar baz\n" * num_samples
+
+    assert "'padded_vocab_size': 512, 'n_layer': 2, 'n_head': 4, 'n_embd': 8" in err.getvalue()
+
+
+def test_lora_variables_exist():
```
Comment (carmocca) [inline tests/test_generate_lora.py:92]: Isn't this test covered already by the assertions at `test_lora_is_enabled`?

**Review by carmocca — COMMENTED** [review]
Thank you! This is useful

File: tests/test_generate_lora.py (hunk 2 above)
Comment (Andrei-Aksionov) [reply tests/test_generate_lora.py:16]: Damn, I wasn't attentive enough. Sorry for that.
Indeed `generate.lora` uses `generate.base:generate`.

File: tests/test_generate_lora.py (hunk 3 above)
Comment (Andrei-Aksionov) [reply tests/test_generate_lora.py:61]: The logic is that `test_lora_is_enabled` checks that LoRA is applied to at least one layer (query, key, value, projections, mlp, head), while `test_lora_variables_exist` checks that those global variables are provided (r, alpha, dropout, ...).
Ideally to check that instance of `Config` class contains these arguments, but couldn't figure out how to do it. Unfortunately not very proficient with mocks 😞.

Btw, do you have any ideas what else LoRA-specific I can check?

**Review by awaelchli — APPROVED** [review]

**Pull request merged**